articles/avoid-fix-flaky-tests.md

```diff
@@ -0,0 +1,73 @@
+---
+authorName: Adrian Herzog
+authorGithubUsername: adiherzog
+issue: 42
+title: How to avoid / deal with flaky system tests
+---
+# {{page.title}}
+
+There are many reasons to avoid automated tests that check an entire system or even a group of connected systems. But there are also good reasons to build and maintain a reasonable number of such tests. For this article, let's assume that the tests are there for a good reason and provide valuable information to the team.
+
+Now, what can you do if one or more of these tests are flaky? By flaky I mean that the test changes from green to red or vice versa without any changes to the code.
+
+Here are a number of strategies that hopefully will help you keeping the tests stable.
+
+
+# Increase Your Chances To Succeed
+
+## Only Write Automated Tests For What's Under Your Control
+
+If the system under tests (SUT) containts components which are not under your control, it's very hard to keep the SUT in a stable state. For those systems not under your control, you might not know whether they have changed in the meantime or not. And even if you do know, you are most probably not able to influence them and change them back to how you need them.
```

Comment:
There is a discussion on defining interfaces/contracts to 3rd-party systems in here. 
You are also making an implicit assumption of live systems (as in a set of services), which while prevalent, is not global. 
There is the case of deliverables like 3rd party libraries that are bound in the code - while the advice does not change, it is easier to enforce as you control the points in time where integration happens.
You need a process that tests against the interface and warns you of breaking changes and this makes sense as an automated test, so the advice should be to cleanly separate the two and not mix interfaced systems in your tests.

I also would not start with this as the first advice - Maybe you should order them by significance (Broken Windows first of all etc.)

Reply:
Good point. I change this paragraph to "Control as much as possible" and give some examples of things that should be kept under control.

Reply:
Moving broken windows up